docs/resources/server/enchantments.md

```diff
@@ -0,0 +1,85 @@
+# Enchantments
+
+Enchantments are enchancements that can be applied to tools and other items. As of 1.21, enchantments are stored on items as [Data Components], are defined in JSON, and are comprised of Enchantment Components.
+
+A new enchantment can be added by creating a JSON file in your namespace's `enchantment` datapack subfolder. For example, to create an enchantment called `examplemod:example_enchant`, one could create the file `data/examplemod/enchantment/example_enchantment.json`. Enchantments added this way are automatically registered into the game as part of the data pack loading process.
+
+The JSON format for Enchantments is summarized on the [Enchantment definition Minecraft wiki page].
```

Comment:
This should also be properly explained here.

Reply:
By this, do you mean I should copy/paraphrase the JSON format section from the Minecraft wiki?

Reply:
More or less. It shouldn't be a straight copy and should have more details related to how it works from a modder's perspective rather than a modpack makers'. It would also be good to add a datagen equivalent, similar to what's done in the biome modifiers PR.

Reply:
I've added the JSON schema, but am not sure about how datagen works for enchantments.

Reply:
It's the same way as any other datapack registry. You are simply creating the object within a registry set builder like so: https://docs.neoforged.net/docs/worldgen/biomemodifier#datagenning-biome-modifiers